plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeHistoryTable.java

```diff
@@ -0,0 +1,138 @@
+import io.trino.spi.type.TypeManager;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.TimestampWithTimeZoneType.TIMESTAMP_TZ_MILLIS;
+import static io.trino.spi.type.TypeSignature.mapType;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Comparator.comparingLong;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeHistoryTable
+        implements SystemTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final List<CommitInfoEntry> commitInfoEntries;
+
+    public DeltaLakeHistoryTable(SchemaTableName tableName, List<CommitInfoEntry> commitInfoEntries, TypeManager typeManager)
+    {
+        requireNonNull(typeManager, "typeManager is null");
+        this.commitInfoEntries = ImmutableList.copyOf(requireNonNull(commitInfoEntries, "commitInfoEntries is null")).stream()
+                .sorted(comparingLong(CommitInfoEntry::getVersion).reversed())
+                .collect(toImmutableList());
+
+        tableMetadata = new ConnectorTableMetadata(
+                requireNonNull(tableName, "tableName is null"),
+                ImmutableList.<ColumnMetadata>builder()
+                        .add(new ColumnMetadata("version", BIGINT))
+                        .add(new ColumnMetadata("timestamp", TIMESTAMP_TZ_MILLIS))
+                        .add(new ColumnMetadata("user_id", VARCHAR))
+                        .add(new ColumnMetadata("user_name", VARCHAR))
+                        .add(new ColumnMetadata("operation", VARCHAR))
+                        .add(new ColumnMetadata("operation_parameters", typeManager.getType(mapType(VARCHAR.getTypeSignature(), VARCHAR.getTypeSignature()))))
+                        .add(new ColumnMetadata("cluster_id", VARCHAR))
+                        .add(new ColumnMetadata("read_version", BIGINT))
+                        .add(new ColumnMetadata("isolation_level", VARCHAR))
+                        .add(new ColumnMetadata("is_blind_append", BOOLEAN))
+                        //TODO add support for operationMetrics, userMetadata, engineInfo
+                        .build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        if (commitInfoEntries.isEmpty()) {
+            return new FixedPageSource(ImmutableList.of());
+        }
+        return new FixedPageSource(buildPages(session));
+    }
+
+    private List<Page> buildPages(ConnectorSession session)
```

Comment:
Similarly to what @krvikash is working on for Iceberg, we should avoid materializing all the entries here and do a streaming approach instead.

Reply:
We could do this eventually as a follow-up - in case it is needed.